Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp

```diff
@@ -25,13 +25,25 @@ namespace AZ
         {
             static AZ::Name GetMapName(const DecalMapType mapType)
             {
-                // Using local static to avoid cost of creating AZ::Name. Also so that this can be called from other static functions
-                static AZStd::array<AZ::Name, DecalMapType_Num> mapNames =
+                // TODO: We can change this back to using static names (if needed) after static allocations are supported
+                // See https://github.com/o3de/sig-core/pull/31
+                // 
+                //static AZStd::array<AZ::Name, DecalMapType_Num> mapNames =
+                //{
+                //    AZ::Name("baseColor.textureMap"),
+                //    AZ::Name("normal.textureMap")
+                //};
+                //return mapNames[mapType];
+
```

Comment:
You could reduce the amount of commented out code, by just only commenting out the `static` keyword.

Reply:
The new implementation is shorter, cleaner, and probably safer because of the switch. Just delete the old commented code and add a todo to make them static in the future.